### PR TITLE
refactor projects list

### PR DESCRIPTION
**Pull request by Haokun-Chen**
##### SUMMARY

- Refactor Project List to use new list components

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### ADDITIONAL INFORMATION

- "Last Modified" and "Last Used" fields in the project list still needs api update to provide modified_by and launch_by in the summary_fields to show the user and link

- Change the status section in at-row-item directive and corresponding styles. This also fixes a bug on the job status icon tooltip. Previously, the job status icon doesn't show tooltip even though tooltip info is implemented. 

- Tooltip is not added for action buttons for now, considering we want to refactor those action buttons later

### CODE REVIEW

File: awx/ui/client/features/projects/projectsList.controller.js [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    'GetProjectToolTip',
+    'resolvedModels',
+    'ProjectsStrings',
+    'Wait',
+    'ngToast',
+    'Prompt',
+    'GetBasePath',
+    'QuerySet',
+    'ProjectUpdate',
+];
```
Comment (jakemcdermott) [inline awx/ui/client/features/projects/projectsList.controller.js:437]: I think it is worth the investment to try to remove as many of these dependencies as we can. Are any of these easily removable, replaceable with helper functions, etc.?

Comment (jakemcdermott) [reply awx/ui/client/features/projects/projectsList.controller.js:437]: `GetProjectIcon` and `GetProjectToolTip` might be easy targets here - https://github.com/ansible/awx/search?q=GetProjectIcon&unscoped_q=GetProjectIcon

Maybe this logic could live in some sort of component?

Comment (Haokun-Chen) [reply awx/ui/client/features/projects/projectsList.controller.js:437]: Yes you're right... I found the original buildtoolTip logic to be confusing...

File: awx/ui/client/features/index.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ import atFeaturesTemplates from '~features/templates';
 import atFeaturesUsers from '~features/users';
 import atFeaturesJobs from '~features/jobs';
```
Comment (jlmitch5) [inline awx/ui/client/features/index.js]: remove space

File: awx/ui/client/features/projects/index.controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            vm.count = count;
+        }
+    });
+}
+
+IndexProjectsController.$inject = [
+    '$scope',
+    'ProjectsStrings',
+    'Dataset',
+    'resolvedModels'
```
Comment (jlmitch5) [inline awx/ui/client/features/projects/index.controller.js]: models and canAdd are concerns at the list view level, not the index view (which is just the outer container of the list).  Move that stuff to projectsList.controller.js

**Comment by wenottingham** [comment]
Apologies, this now has conflicts. Can you rebase and fix the conflicts?

**Comment by Haokun-Chen** [comment]
Sure. Checking the conflicts and rebasing

File: awx/ui/client/lib/components/list/row-item.partial.html [hunk 4]
```diff
@@ -1,12 +1,11 @@
 <div class="at-RowItem" ng-class="{'at-RowItem--isHeader': headerValue, 'at-RowItem--inline': inline}"
     ng-show="status || headerValue || value || valueBindHtml || (smartStatus && smartStatus.summary_fields.recent_jobs.length) || (tagValues && tagValues.length)">
         <div class="at-RowItem-status" ng-if="status">
-            <a ng-if="headerLink" ng-href="{{ headerLink }}"
-                aw-tool-tip="{{ statusTip }}" aw-tip-watch="statusTip"
-                data-placement="top">
-                    <i class="fa icon-job-{{ status }}"></i>
+            <a ng-if="statusTip" ng-click="statusClick()">
```
Comment (jlmitch5) [inline awx/ui/client/lib/components/list/row-item.partial.html]: this should be an "a" tag if it doesn't link to something.  you either want div or button

**Review by jlmitch5 — CHANGES_REQUESTED** [review]
great job!

**Review by jlmitch5 — APPROVED** [review]
very nice!

**Pull request merged**